GOVERNANCE.md

```diff
@@ -4,83 +4,139 @@ This document defines project governance for the Strimzi project.
 
-## Voting
+## Voting rules
 
 The Strimzi project employs voting to ensure no single member can dominate the project.
 
 For formal votes, a specific statement of what is being voted on should be added to the relevant GitHub issue or PR.
 Voters should indicate their yes/no vote on that issue or PR, and after a suitable period of time, the votes will be tallied and the outcome noted.
 
-### Approving PRs
+The project uses different kinds of vote for different purposes.
+
+* PR approval votes
+* Lazy majority votes
+* Explicit majority votes
+
+Who may vote, where votes are recorded and the duration of the voting also depends on the subject of vote, as explained in the following sections.
 
-PRs may be merged after receiving at least two positive maintainer votes.
+### PR approval rules
+
+PRs against most repositories may be merged after receiving at least two positive maintainer votes.
 Voting for PRs can be done using a comment in the PR or by approving (or requesting changes) the PR.
 If the PR author is a maintainer, this counts as a vote.
 
-PRs which affect only a single component can be approved by at least one positive maintainer vote and one positive vote from an owner of given component.
+PRs which affect only a single component can be approved by at least one positive maintainer vote and one positive vote from an owner of the given component.
 If the PR author is a maintainer or an owner of given component, this counts as a vote.
 
-### Proposals
-
-Proposals can be opened against the [proposals repository](https://github.com/strimzi/proposals).
-Proposals should cover any changes to the Strimzi project which might significantly impact its users or the project direction.
+### Lazy majority rules
 
-Voting about proposals is using +1, 0, -1 votes and their fractions, where:
+Lazy majority votes use +1, 0, -1 votes and their fractions, where:
 * +1 means "yes"
 * -1 means "no"
 * The numbers between +1 and -1 indicate how strongly you feel about the proposal
 
 _(Inspired by [Apache Software Foundation voting](https://www.apache.org/foundation/voting.html#expressing-votes-1-0-1-and-fractions))_
 
-Votes from maintainers are binding and count towards the approval of the proposal.
-Non-maintainers are allowed and encouraged to vote as well.
-But their votes are non-binding and do not count towards approval of the proposal.
+Binding votes are those cast by people in specific roles defined for that vote and count towards the approval of the subject of the vote.
+Non-binding may be cast by people not in any of those roles and do not count towards approval of the subject of the vote.
 
-+1 votes can be expressed by approving the proposal PR or in the comments.
-Other votes should be expressed in the comments.
-For example "+1 (binding)", or for a non-maintainer "+1 (non-binding)".
+The vote succeeds when there are least three +1 binding votes and no -1 binding votes.
```

Comment:
so just one -1 can screw up a majority of, for example, five +1 from other maintainers? I know it was already here for proposals but are we sure it works?

Reply:
> so just one -1 can screw up a majority of, for example, five +1 from other maintainers? 

Yes.

> I know it was already here for proposals but are we sure it works?

Answering very literally: We don't really have enough data to answer that meaningfully. Even if then I don't know how you'd objectively decide on what "it works" means. It clearly is a mechanism which can a plurality from getting their way. In that sense it clearly doesn't "work". But the end objective is: Does it make for better decisions than some alternative?

But I think you mean: "Are we happy with the implication that every maintainer holds a veto?" Presumably we _were_ happy with that once, because it's in our current rules. So what's changed?

More generally, I think it's easy to poke holes in any system. We're not going to get a perfect system with this revision. The important things are:

1. we have a mechanism to evolve the rules as circumstances change
2. we think this proposed change is an improvement on what went before

Reply:
Shame on me but I think this was another point I forgot about the current rules (like the one I can vote 0.4 ;-)), maybe just because it never happened in the reality.
But in general, I think that in a community, it's not that great having every maintainer holding a veto. I think we still would need a majority. If just a single maintainer doesn't think the proposal is good but the rest of maintainers agree, I don't think that just that -1 should stop the overall approval. This could just get a proposal stuck, because that maintainer is asking for changes (so the -1) which others don't agree and these changes will never be applied.

Reply:
TBH I think I'd rather see if we can get these changes through without opening up a larger debate about the veto. That's because:

* It's not a problem we've seen in practice.
* I do think there's value in getting a consensus. The veto forces people to address the concerns of others, rather than dismissing them. 
* I think there are social dynamics which usually tend to prevent a single maintainer from using their veto unnecessarily or in a problematic way.
* If it did become a problem, there is still the option of removing a maintainer.

Feel free to open a separate discussion about it if you feel strongly about it.

Reply:
> I do think there's value in getting a consensus. The veto forces people to address the concerns of others, rather than dismissing them.

Why people should address concerns of others if "others" are stuck on their positions and thanks to the veto they can stop a proposal? This is my big fear, even if as you said it has not been a problem in practice until today. But how are we going to deal with it if it happens? Removing a maintainer because he is stuck on his/her position and blocking a proposal while all other maintainers are in agreement? Tbh I don't see it that great.